### ISSUE TITLE
torch.onnx.export creates unnecessary padding node for average pooling.

### ISSUE THREAD

**Issue by leimao** [pytorch/pytorch#56218]
## 🐛 Bug

ONNX export for the following dummy model contains unnecessary `Pad` op.

```
import torch
import torch.nn as nn

input_tensor = torch.rand(1, 3, 224, 224)
opset_version = 10

# Dummy model

class DummyModel(nn.Module):

    def __init__(self):

        super(DummyModel, self).__init__()

        self.avg_pool2d = nn.AvgPool2d(kernel_size=(2, 2), stride=(2, 2))

    def forward(self, x):

        x = self.avg_pool2d(x)

        return x

dummy_model = DummyModel()
dummy_model.eval()
onnx_file_path = "dummy_model.onnx"
torch.onnx.export(model=dummy_model,
                    args=input_tensor,
                    f=onnx_file_path,
                    do_constant_folding=True,
                    opset_version=opset_version,
                    export_params=True,
                    )
```

![image](https://user-images.githubusercontent.com/17606112/114954545-b008d900-9e0f-11eb-9aff-2f8ce901fb15.png)

## To Reproduce

Running using NVIDIA PyTorch container `nvcr.io/nvidia/pytorch:21.02-py3`.

## Expected behavior

I will expect there is no `Pad` op as there is no padding used for average pooling during runtime.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

PyTorch version: 1.8.0a0+52ea372
Is debug build: False
CUDA used to build PyTorch: 11.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.1 LTS (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.19.4

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 11.2.67
GPU models and configuration: GPU 0: TITAN V
Nvidia driver version: 460.32.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.0
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] pytorch-transformers==1.1.0
[pip3] torch==1.8.0a0+52ea372
[pip3] torchtext==0.9.0a0
[pip3] torchvision==0.9.0a0
[conda] magma-cuda110             2.5.2                         5    local
[conda] mkl                       2019.4                      243  
[conda] mkl-include               2019.4                      243  
[conda] nomkl                     3.0                           0  
[conda] numpy                     1.19.2           py38h6163131_0  
[conda] numpy-base                1.19.2           py38h75fe3a5_0  
[conda] pytorch-transformers      1.1.0                    pypi_0    pypi
[conda] torch                     1.8.0a0+52ea372          pypi_0    pypi
[conda] torchtext                 0.9.0a0                  pypi_0    pypi
[conda] torchvision               0.9.0a0                  pypi_0    pypi

## Additional context

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @SplitInfinity

**Comment by nwhitehead** [comment]
I also see this issue.

To add a bit to the bug, the extra padding operation shows up in the ONNX export when `count_include_pad=True` is passed to `nn.AvgPool2d` (the PyTorch default). It does not show up when `count_include_pad=False` is passed.

**Comment by shubhambhokare1** [comment]
Hi @leimao,

We’ve gone ahead and closed this issue because it is stale.
If you still believe this issue is relevant, please feel free to reopen the issue and we will triage it as necessary. Please specify in a comment any updated information you may have so that we can address it effectively. We encourage you to try the latest pytorch-preview (nightly) version to see if it has resolved the issue.

Thanks,
ONNX Converter team
